monzo.js

```diff
@@ -1,20 +1,83 @@
 var PD = require("probability-distributions");
 
-var goal = 25000;
-var pledgeValues = [10,20,50,100,250,500,1000];
+var goal = 2500;
+var pledgeValues = [10,20,50,100];
 var maxPledgeValue = pledgeValues[pledgeValues.length - 1];
-var dataset = PD.sample(pledgeValues, 1000, true, [7,6,5,4,3,2,1]);
+var dataset = PD.sample(pledgeValues, 1000, true, [7,6,5,4]);
 
 var datasetTotal = dataset.reduce((a, b) => a + b, 0);
-var randomlyChosenPledge;
-while (datasetTotal >= goal ) {
-	randomlyChosenPledge = dataset.splice(
-    Math.floor(Math.random() * dataset.length), 1
-  )[0];
-	datasetTotal -= randomlyChosenPledge;
+
+/*  Convert array to object so each pledge has an ID */
+dataset = dataset.reduce(function(obj, currentValue, index) {
+    obj[index] = currentValue;
+    return obj;
+}, {});
+
+/*  Make copy of original dataset and total and randomly select pledges to
+    to remove so it now contains only chosen pledges */
+var chosenDatasetTotal = datasetTotal;
+var chosenDataset = {};
+for (var i in dataset) chosenDataset[i] = dataset[i]; // replace with Object.assign()?
+while (chosenDatasetTotal >= goal ) {
+  var ids = Object.keys(chosenDataset);
```

Comment:
`var randomId = ids[Math.floor(Math.random() * ids.length)]` shouldn't be indented.